### PR TITLE
Feature/active snippets

### PR DESCRIPTION
**Pull request by gtluszcz**
This pull adds activeness of snippets.
From now on you will be able to deactivate certain snippet and activate it again!

zrzut ekranu 2018-10-13 o 23 54 13

### CODE REVIEW

File: src/main/modules/SnippetsManager.js [hunk 1]
```diff
@@ -170,6 +181,10 @@ class SnippetsManager {
 
     async _replaceSnippetIfMatchFound() {
         for (const snippet of this.store.get('snippets')) {
+            if (!snippet.active) {
+                continue
```
Comment (DCzajkowski) [inline src/main/modules/SnippetsManager.js:185]: 👍

File: src/main/modules/SnippetsManager.js [hunk 2]
```diff
@@ -28,6 +28,17 @@ class SnippetsManager {
         this.keyboardHandler.start()
 
         fixPath()
+        this._addNewFields()
+    }
+
+    _addNewFields() {
+        const newSnippets = this.store.get('snippets').map(s => {
+            s.active = s.active || true
+            s.regex = s.regex || false
```
Comment (DCzajkowski) [inline src/main/modules/SnippetsManager.js]: These won't work. If someone sets `active` to `false`, you would map it to true 😛

File: src/main/modules/SnippetsManager.js [hunk 3]
```diff
@@ -28,6 +28,17 @@ class SnippetsManager {
         this.keyboardHandler.start()
 
         fixPath()
+        this._addNewFields()
+    }
+
+    _addNewFields() {
+        const newSnippets = this.store.get('snippets').map(s => {
```
Comment (DCzajkowski) [inline src/main/modules/SnippetsManager.js]: I'd personally either inline this variable or just call it `const snippets`. Just a personal preference.

File: src/main/modules/SnippetsManager.js [hunk 4]
```diff
@@ -28,6 +28,17 @@ class SnippetsManager {
         this.keyboardHandler.start()
 
         fixPath()
+        this._addNewFields()
+    }
+
+    _addNewFields() {
```
Comment (DCzajkowski) [inline src/main/modules/SnippetsManager.js]: `_addNewFields` -> `_ensureAllFieldsArePresent`? "Add new fields" implies new fields are always added, where in fact the function is just ensuring everything is set properly.

File: src/renderer/windows/main/components/Snippets.vue [hunk 5]
```diff
@@ -82,29 +83,54 @@
...
+                                        v-else
+                                        :checked="false"
+                                        class="fill-current w-6 h-6 rounded-sm group-focus:shadow-outline"
+                                        :class="['text-grey-lightest', 'text-grey-darkest'][theme]"
+                                    ></icon-checkbox>
+                                </div>
 
-                            Use Regular Expression
-                        </label>
+                                active
```
Comment (DCzajkowski) [inline src/renderer/windows/main/components/Snippets.vue]: "active" -> "Active" to preserve Title Case style.

**Review by DCzajkowski — COMMENTED** [review]
Are you planning to add tests for these features? We should probably require tests on all future PRs and invest some more time in heavy testing.

File: src/main/modules/SnippetsManager.js (hunk 2 above)
Comment (DCzajkowski) [reply src/main/modules/SnippetsManager.js]: The `=== undefined` check would be appropriate here

File: src/renderer/windows/main/components/Snippets.vue (hunk 5 above)
Comment (gtluszcz) [reply src/renderer/windows/main/components/Snippets.vue]: good catch

File: src/main/modules/SnippetsManager.js (hunk 2 above)
Comment (gtluszcz) [reply src/main/modules/SnippetsManager.js]: omg i am so embarrassed right now xd

File: src/main/modules/SnippetsManager.js (hunk 4 above)
Comment (gtluszcz) [reply src/main/modules/SnippetsManager.js]: okay fine 👍

File: src/main/modules/SnippetsManager.js (hunk 3 above)
Comment (gtluszcz) [reply src/main/modules/SnippetsManager.js]: ok

File: src/main/modules/SnippetsManager.js (hunk 1 above)
Comment (gtluszcz) [reply src/main/modules/SnippetsManager.js:185]: 😄

**Comment by gtluszcz** [comment]
I don't really have experience with unit tests in js. To be honest i wasn't gonna do that, although i see clearly that we are missing tests and we should start to write them. I have to catch up on this fast!

**Pull request merged**